### PR TITLE
add fsthttp.Transport, which is an http.RoundTripper implementation

### PR DESCRIPTION
**Pull request by joeshaw**
This allows existing Go code which uses an http.Client to be adapted to
work on Compute@Edge.

This is intended only to adapt existing code.  Using http.Client pulls
in substantially more code than is needed for a simple request,
resulting in slower compile times and larger binaries.  We recommend new
code use the fsthttp.Request type and its Send() method directly
whenever possible.

### CODE REVIEW

**Review by dgryski — APPROVED** [review]
LGTM.

**Comment by joeshaw** [comment]
@dgryski PTAL

File: fsthttp/transport.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	return &Transport{
+		defaultBackend: backend,
+		backends:       make(map[string]string),
+	}
+}
+
+// AddHostBackend adds a new host-to-backend mapping.  Multiple hosts
+// may be mapped to the same backend.
+func (t *Transport) AddHostBackend(host, backend string) {
+	t.backends[strings.ToLower(host)] = backend
```
Comment (dgryski) [inline fsthttp/transport.go:47]: Do we want to protect `t.backends` with a mutex?

**Review by dgryski — APPROVED** [review]
LGTM.  Mutex addition is optional.

File: fsthttp/transport.go (hunk 1 above)
Comment (joeshaw) [reply fsthttp/transport.go:47]: I don't have a strong opinion on this.  It seems unlikely to me that users would add mappings concurrently, so my vote is to wait until it becomes a problem in practice.

**Pull request merged**